sys/newlib/Makefile.include

```diff
@@ -4,6 +4,9 @@ ifneq (,$(filter newlib_nano,$(USEMODULE)))
   # Test if nano.specs is available
   ifeq ($(shell $(LINK) -specs=nano.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
     USE_NEWLIB_NANO = 1
+        ifeq ($(shell echo "int main(){}; void _exit(int n) {while(1);}" | LC_ALL=C $(CC) -xc - -o /dev/null -lc -specs=nano.specs 2>&1 | grep -q "use of wchar_t values across objects may fail" ; echo $$?),0)
```

Comment:
Let's fix the pedantic warnings too:
```
% echo "int main(){}; void _exit(int n) {while(1);}" | LC_ALL=C arm-none-eabi-gcc -xc - -o /dev/null -lc -specs=nano.specs -Wall -Wextra -pedantic
<stdin>:1:13: warning: ISO C does not allow extra ';' outside of a function [-Wpedantic]
<stdin>: In function '_exit':
<stdin>:1:30: warning: unused parameter 'n' [-Wunused-parameter]
```
`int main(){} void _exit(int n) {(void) n;while(1);}` should do it.